main.go

```diff
@@ -85,3 +108,51 @@ type KVResult struct {
 	Key   cloudflare.StorageKey
 	Value string
 }
+
+type kvClient interface {
+	WriteWorkersKVEntries(ctx context.Context, rc *cloudflare.ResourceContainer, params cloudflare.WriteWorkersKVEntriesParams) (cloudflare.Response, error)
+}
+
+var jsRx = regexp.MustCompile(`^\s*["]?\s*{.*`)
+
+func fixStringEncodedJson(ctx context.Context, client kvClient, accountContainer *cloudflare.ResourceContainer, data []byte, namespace string, metadata cloudflare.StorageKey) ([]byte, error) {
+	data, changed := fixStringEncodedJsonOnly(data)
+	if changed {
+		response, err := client.WriteWorkersKVEntries(ctx, accountContainer,
+			cloudflare.WriteWorkersKVEntriesParams{
+				NamespaceID: namespace,
+				KVs: []*cloudflare.WorkersKVPair{
+					{
+						Key:        metadata.Name,
+						Value:      string(data),
+						Expiration: metadata.Expiration,
+						Metadata:   metadata.Metadata,
+					},
+				},
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error writing kv entry:%w", err)
+		}
+		if !response.Success {
+			return nil, fmt.Errorf("uncessful kv write:%+v", response)
+		}
+	}
+	return data, nil
+}
+
+func fixStringEncodedJsonOnly(data []byte) ([]byte, bool) {
+	var result json.RawMessage
+	// Handle json data which cannot be decoded
+	err := json.Unmarshal(data, &result)
+	var jsError *json.SyntaxError
+	if errors.As(err, &jsError) && jsRx.Match(data) {
+		dStr := strings.ReplaceAll(string(data), `\"`, `"`)
+		dStr = strings.ReplaceAll(dStr, `\\`, `\`)
+		dStr = strings.TrimPrefix(dStr, `"`)
+		dStr = strings.TrimSuffix(dStr, `"`)
+		data = []byte(dStr)
+		return data, true
```

Comment:
Do we expect the data to be json-decodeable by the time we're done? If so, it could be nice to check that it now unmarshals successfully.

Or are we also wanting to fix data that wasn't valid json in the first place?